### PR TITLE
Add fp16 support  for FarthestPointSampler

### PR DESCRIPTION
**Pull request by eamonn-zh**
## Description
implemented the issue  #6456 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [Feature]
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: src/geometry/geometry.cc [hunk 1]
```diff
@@ -29,15 +29,24 @@ void FarthestPointSampler(
...
+
+  // ATEN_FLOAT_TYPE_SWITCH_16BITS(array->dtype, FloatType, XPU, "values", {
+  //   ATEN_ID_TYPE_SWITCH(result->dtype, IdType, {
+  //     ATEN_XPU_SWITCH_CUDA(
+  //         array->ctx.device_type, XPU, "FarthestPointSampler", {
+  //           impl::FarthestPointSampler<XPU, FloatType, IdType>(
+  //               array, batch_size, sample_points, dist, start_idx, result);
+  //         });
+  //   });
+  // });
```
Comment (frozenbugs) [inline src/geometry/geometry.cc:49]:
```suggestion
```

File: src/geometry/geometry.cc [hunk 2]
```diff
@@ -29,15 +29,24 @@ void FarthestPointSampler(
...
+
+  ATEN_XPU_SWITCH_CUDA(array->ctx.device_type, XPU, "FarthestPointSampler", {
     ATEN_ID_TYPE_SWITCH(result->dtype, IdType, {
-      ATEN_XPU_SWITCH_CUDA(
-          array->ctx.device_type, XPU, "FarthestPointSampler", {
-            impl::FarthestPointSampler<XPU, FloatType, IdType>(
-                array, batch_size, sample_points, dist, start_idx, result);
-          });
+      ATEN_FLOAT_TYPE_SWITCH_16BITS(array->dtype, FloatType, XPU, "values", {
+        impl::FarthestPointSampler<XPU, FloatType, IdType>(array, batch_size, sample_points, dist, start_idx, result);
```
Comment (frozenbugs) [inline src/geometry/geometry.cc:36]: line too long, max 100 chars.

File: src/geometry/geometry.cc (hunk 1 above)
Comment (frozenbugs) [reply src/geometry/geometry.cc:49]: If we don't need this code, remove it.

File: include/dgl/aten/macro.h [hunk 3]
```diff
@@ -124,7 +124,7 @@
       typedef double FloatType;                                             \
       { __VA_ARGS__ }                                                       \
     } else {                                                                \
-      LOG(FATAL) << (val_name) << " can only be float32 or float64";        \
+      LOG(FATAL) << (val_name) << " can only be float32, float64 or float16";\
```
Comment (frozenbugs) [inline include/dgl/aten/macro.h:127]:
```suggestion
      LOG(FATAL) << (val_name)                                                             \
                            << " can only be float16, float32 or float64";          \
```

**Comment by eamonn-zh** [comment]
it breaks the CPU version implementation.

**Pull request closed without merging**